lib/diagnostic.js

```diff
@@ -8,40 +8,129 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-const min = function min() {
+
+let arr = [1,2,3];
+
+
+let callbackMin = function (element) {
+
+  let firstElement = arr[0];
+
+  for (let i = 0; i < array.length; i++) {
+    let result =   array[i];
+
+if (firstElement < result) {
+  return firstElement;
+}
+
+  }
+
+};
+
+const min = function min(array, callback) {
+
 
 };
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
-const sum = function sum() {
+const sum = function sum(prev, curr) {
+
+  return prev + sum;
 
 };
 
+const reduce = function(arr, callback, init) {
+
+  let result;
+
+  if(init!=undefined){
+    result = init;
+  }
+
+  for (let i = 0; i < array.length; i++) {
+    result = callback(result,array[i]);
+  }
+
+};
+
+reduce(arr, sum, 0);
+
+
+
 // Question 3
 // Write a function that returns true
 // if invoking the callback for any element
 // in the array returns a truthy value
+
```

Comment:
So for question 3, you'd just want to loop over an array and if any elements applied to a callback return `true`, the function can return `true`.

```
const any = function any(array, callback) {
  for (let i = 0; i < array.length; i++) {
    if (callback(array[i])) {
      return true;
    }
  }

  return false;
};
```